### PR TITLE
Docu for gtest and dflags.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

**Comment by dkorolev** [comment]
@deathbaba @sompylasar PTAL.

**Comment by dkorolev** [comment]
https://github.com/KnowSheet/Bricks/issues/78

File: 3party/gtest/docu/docu_99gtest.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+5. **Exceptions**
+  
+   `ASSERT_THROW(statement, exception_type);` ensures the exception is thrown.
+
+6. **Death Tests**
+  
+   `ASSERT_DEATH(function(), "Expected regex for the last line of standard error.");`
+  
+   can be used to ensure certain call fails. The convention is to use the `"DeathTest"` suffix for those tests and to not mix functional tests with death tests.
```
Comment (sompylasar) [inline 3party/gtest/docu/docu_99gtest.md]: Please add some detail what is a fail here. An exception or ...?

File: 3party/gtest/docu/docu_99gtest.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+7. **Templated Tests**
+  
+   `gtest` supports templated tests, where objects of various tests are passed to the same test method.
+  
+   Each type results in the whole new statically compiled test.
```
Comment (sompylasar) [inline 3party/gtest/docu/docu_99gtest.md]: Example?

File: 3party/gtest/docu/docu_99gtest.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+## Unit Testing: `gtest`
+
+Bricks constains a header-only port of Google's [`GoogleTest`](http://code.google.com/p/googletest/): a great C++ unit testing library open-sourced by [**Google**](https://www.google.com/finance?q=GOOG).
+
+A three-minute intro:
+
+1. **Logic**
+  
+  `ASSERT_*` interrupts the `TEST() { .. }` if failing,
```
Comment (sompylasar) [inline 3party/gtest/docu/docu_99gtest.md]: `0. **Structure**`
about `TEST()`s themselves.

File: 3party/gtest/docu/docu_99gtest.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+## Unit Testing: `gtest`
+
+Bricks constains a header-only port of Google's [`GoogleTest`](http://code.google.com/p/googletest/): a great C++ unit testing library open-sourced by [**Google**](https://www.google.com/finance?q=GOOG).
```
Comment (sompylasar) [inline 3party/gtest/docu/docu_99gtest.md]: `constains` -> `contains`

**Comment by sompylasar** [comment]
@dkorolev Please look through my comments. Thanks.

File: 3party/gtest/docu/docu_99gtest.md (hunk 4 above)
Comment (dkorolev) [inline 3party/gtest/docu/docu_99gtest.md]: Fixed.

File: 3party/gtest/docu/docu_99gtest.md (hunk 3 above)
Comment (dkorolev) [inline 3party/gtest/docu/docu_99gtest.md]: Added a link to an existing test.

File: 3party/gtest/docu/docu_99gtest.md (hunk 1 above)
Comment (dkorolev) [inline 3party/gtest/docu/docu_99gtest.md]: Added.

File: 3party/gtest/docu/docu_99gtest.md (hunk 2 above)
Comment (dkorolev) [inline 3party/gtest/docu/docu_99gtest.md]: Not sure I'd like to get into it before at least one external user asks :)

Comment (sompylasar) [inline 3party/gtest/docu/docu_99gtest.md]: OK. I thought we used these in our tests.

**Comment by dkorolev** [comment]
@sompylasar Thanks for the review! PTAL.

File: 3party/gtest/docu/docu_99gtest.md (hunk 2 above)
Comment (dkorolev) [inline 3party/gtest/docu/docu_99gtest.md]: We do, but it still looks like too much trouble to get those involved/linked from here.

File: 3party/gtest/docu/docu_99gtest.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+## Unit Testing: `gtest`
+
+Bricks contains a header-only port of Google's [`GoogleTest`](http://code.google.com/p/googletest/): a great C++ unit testing library open-sourced by [**Google**](https://www.google.com/finance?q=GOOG).
+
+Check out [`Bricks/file/test.cc`](https://github.com/KnowSheet/Bricks/blob/master/file/test.cc), or most of other `test.cc` files in `Bricks` for exampl usage.
```
Comment (sompylasar) [inline 3party/gtest/docu/docu_99gtest.md]: `for exampl usage.` -> `for example usage.`

File: 3party/gtest/docu/docu_99gtest.md (hunk 2 above)
Comment (sompylasar) [inline 3party/gtest/docu/docu_99gtest.md]: OK.

File: 3party/gtest/docu/docu_99gtest.md (hunk 3 above)
Comment (sompylasar) [inline 3party/gtest/docu/docu_99gtest.md]: Thanks.

**Comment by dkorolev** [comment]
Fixed the `exampl`, @sompylasar PTAL :)

**Comment by sompylasar** [comment]
@dkorolev LGTM!

**Comment by dkorolev** [comment]
@sompylasar Thank you!
@deathbaba PTAL.

**Pull request merged**